### PR TITLE
feat: Add logger to preparation

### LINKED ISSUE #34: Add logger and logfile
We should introduce logger and logfile which can be used in debugging and CI.

### PR DESCRIPTION
**Pull request by splasky**
Logger is helpful for generate debug message or info. It can also
disable all debug message when DEBUG flag not set.

### CODE REVIEW

**Comment by jserv** [comment]
It is redundant to denote "Add logger to log message." Logger is always used for logging. Instead, you have to explain how the logger works along with potential integration efforts.

**Comment by jserv** [comment]
Add documentation!

**Comment by splasky** [comment]
related #34

**Comment by jserv** [comment]
Fix conflicts.

**Review by jserv — CHANGES_REQUESTED** [review]
"preparation" should be emphasized since no code snippet is using logger so far.

File: utils/logger.c [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#include "logger.h"
+static int TA_LOG_LEVEL = LV_INFO;
```
Comment (jserv) [inline utils/logger.c:2]: You should implement non-trivial logger, which takes circular buffer into consideration.

File: utils/logger.h [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * Copyright (C) 2019-2020 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
+#ifndef UTIL_LOGGER_H
+#define UTIL_LOGGER_H
+
```
Comment (jserv) [inline utils/logger.h:11]: Add comments about these level.

**Comment by splasky** [comment]
Will use the tangle-accelerator's logger as default logger.

**Pull request closed without merging**